components/ntp_background_images/browser/ntp_background_images_service.cc

```diff
@@ -6,82 +6,394 @@
+constexpr char kNTPSIManifestFile[] = "photo.json";
+constexpr char kNTPSRManifestFile[] = "data.json";
+constexpr char kNTPSRMappingTableFile[] = "mapping-table.json";
 
-std::string ReadPhotosManifest(const base::FilePath& photos_manifest_path) {
+constexpr char kNTPSRMappingTableComponentPublicKey[] = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAp7IWv7wzH/KLrxx7BKWOIIUMDylQNzxwM5Fig2WHc16BoMW9Kaya/g17Bpfp0YIvxdcmDBcB9kFALqQLxi1WQfa9d7YxqcmAGUKo407RMwEa6dQVkIPMFz2ZPGSfFgr526gYOqWh3Q4h8oN94qxBLgFyT25SMK5zQDGyq96ntME4MQRNwpDBUv7DDK7Npwe9iE8cBgzYTvf0taAFn2ZZi1RhS0RzpdynucpKosnc0sVBLTXy+HDvnMr+77T48zM0YmpjIh8Qmrp9CNbKzZUsZzNfnHpL9IZnjwQ51EOYdPGX2r1obChVZN19HzpK5scZEMRKoCMfCepWpEkMSIoPzQIDAQAB";  // NOLINT
+constexpr char kNTPSRMappingTableComponentID[] =
+    "heplpbhjcbmiibdlchlanmdenffpiibo";
+constexpr char kNTPSRMappingTableComponentName[] =
+    "NTP Super Referral mapping table";
+
+std::string GetMappingTableData(const base::FilePath& installed_dir) {
   std::string contents;
-  bool success = base::ReadFileToString(photos_manifest_path, &contents);
-  if (!success || contents.empty()) {
-    DVLOG(2) << "ReadPhotosManifest: cannot "
-             << "read photo.json file " << photos_manifest_path;
-  }
+  const auto json_path = installed_dir.AppendASCII(kNTPSRMappingTableFile);
+  base::ReadFileToString(json_path, &contents);
   return contents;
 }
 
-NTPBackgroundImagesData* GetDemoWallpaper() {
-  static auto demo = std::make_unique<NTPBackgroundImagesData>();
-  demo->url_prefix = "chrome://newtab/ntp-dummy-brandedwallpaper/";
-  demo->backgrounds = {
-      { base::FilePath(FILE_PATH_LITERAL("wallpaper1.jpg")), { 3988, 2049 } },
-      { base::FilePath(FILE_PATH_LITERAL("wallpaper2.jpg")), { 5233, 3464 } },
-      { base::FilePath(FILE_PATH_LITERAL("wallpaper3.jpg")), {  0, 0 } },
-  };
-  demo->logo_alt_text = "Technikke: For music lovers.";
-  demo->logo_company_name = "Technikke";
-  demo->logo_destination_url = "https://brave.com";
-  return demo.get();
+void CacheSuperReferralData(const std::string& data_json,
+                            const base::FilePath& installed_dir,
+                            const base::FilePath& super_referral_cache_dir) {
+  NTPBackgroundImagesData data(data_json,
+                               super_referral_cache_dir);
+
+  base::CreateDirectory(super_referral_cache_dir);
+  // Cache logo image
+  base::CopyFile(installed_dir.Append(data.logo_image_file.BaseName()),
+                 data.logo_image_file);
+  // Cache topsite favicon images
+  for (const auto& top_site : data.top_sites) {
+    base::CopyFile(installed_dir.Append(top_site.image_file.BaseName()),
+                   top_site.image_file);
+  }
+  // Cache background images
+  for (const auto& background : data.backgrounds) {
+    base::CopyFile(installed_dir.Append(background.image_file.BaseName()),
+                   background.image_file);
+  }
+}
+
+// If registered component is for sponsored images wallpaper, it has photo.json
```

Comment:
I think the components need to have identical structure and file naming

Reply:
Fixed. same file name will be used.